CRM/Pledge/BAO/Pledge.php

```diff
@@ -1202,4 +1202,36 @@ protected static function getTransactionalStatus() {
     return array_diff(array_flip($paymentStatus), self::getNonTransactionalStatus());
   }
 
+  /**
+   * Create recur record for pledge.
+   *
+   */
+  public static function createRecurRecord($pledge, $params, $form) {
```

Comment:
Passing the $form object to the BAO is a really bad pattern we are trying to deprecate Try to figure out a way to avoid it!